src/OpenTelemetry.Api/Trace/ActivityExtensions.cs

```diff
@@ -60,120 +60,7 @@ public static void SetStatus(this Activity activity, Status status)
-        }
-
-        /// <summary>
-        /// Enumerates all the key/value pairs on an <see cref="ActivityLink"/> without performing an allocation.
-        /// </summary>
-        /// <typeparam name="T">The struct <see cref="IActivityEnumerator{T}"/> implementation to use for the enumeration.</typeparam>
-        /// <param name="activityLink">ActivityLink instance.</param>
-        /// <param name="tagEnumerator">Tag enumerator.</param>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "ActivityProcessor is hot path")]
-        public static void EnumerateTags<T>(this ActivityLink activityLink, ref T tagEnumerator)
-            where T : struct, IActivityEnumerator<KeyValuePair<string, object>>
-        {
-            ActivityTagsEnumeratorFactory<T>.Enumerate(activityLink, ref tagEnumerator);
-        }
-
-        /// <summary>
-        /// Enumerates all the <see cref="ActivityEvent"/>s on an <see cref="Activity"/> without performing an allocation.
-        /// </summary>
-        /// <typeparam name="T">The struct <see cref="IActivityEnumerator{T}"/> implementation to use for the enumeration.</typeparam>
-        /// <param name="activity">Activity instance.</param>
-        /// <param name="eventEnumerator">Event enumerator.</param>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "ActivityProcessor is hot path")]
-        public static void EnumerateEvents<T>(this Activity activity, ref T eventEnumerator)
-            where T : struct, IActivityEnumerator<ActivityEvent>
-        {
-            Debug.Assert(activity != null, "Activity should not be null");
-
-            ActivityEventsEnumeratorFactory<T>.Enumerate(activity, ref eventEnumerator);
-        }
-
-        /// <summary>
-        /// Enumerates all the key/value pairs on an <see cref="ActivityEvent"/> without performing an allocation.
-        /// </summary>
-        /// <typeparam name="T">The struct <see cref="IActivityEnumerator{T}"/> implementation to use for the enumeration.</typeparam>
-        /// <param name="activityEvent">ActivityEvent instance.</param>
-        /// <param name="tagEnumerator">Tag enumerator.</param>
-        [MethodImpl(MethodImplOptions.AggressiveInlining)]
-        [System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1062:Validate arguments of public methods", Justification = "ActivityProcessor is hot path")]
-        public static void EnumerateTags<T>(this ActivityEvent activityEvent, ref T tagEnumerator)
-            where T : struct, IActivityEnumerator<KeyValuePair<string, object>>
-        {
-            ActivityTagsEnumeratorFactory<T>.Enumerate(activityEvent, ref tagEnumerator);
+            return activity.GetStatusHelper();
```

Comment:
Might want to add `[MethodImpl(MethodImplOptions.AggressiveInlining)]`.

Reply:
just realized its already there, right?

Reply:
Ah sorry it was above what the diff was showing.